### PR TITLE
Moving Kubelet --node-status-max-images flag to configuration

### PR DESCRIPTION
**Pull request by knabben**
**What type of PR is this?**

/kind cleanup
/area kubelet
/area kubelet-api

**What this PR does / why we need it**:

Moving --node-status-max-images to Kubelet configuration

**Which issue(s) this PR fixes**:

Refs #86843 #90747

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
The Kubelet's --node-status-max-images option is now available via the Kubelet config file field nodeStatusMaxImage
```

### CODE REVIEW

**Comment by knabben** [comment]
Graduating --node-max-status-image and copying the flag to the configuration file

/assign @mtaufen @wojtek-t

File: pkg/kubelet/apis/config/v1beta1/defaults.go [hunk 1]
```diff
@@ -176,6 +176,9 @@ func SetDefaults_KubeletConfiguration(obj *kubeletconfigv1beta1.KubeletConfigura
 	if obj.CPUCFSQuotaPeriod == nil {
 		obj.CPUCFSQuotaPeriod = &metav1.Duration{Duration: 100 * time.Millisecond}
 	}
+	if obj.NodeStatusMaxImages == 0 {
```
Comment (dashpole) [inline pkg/kubelet/apis/config/v1beta1/defaults.go:179]: Could I have previously set the max images to 0 to make the node publish no images?  Is there a way we can preserve that and distinguish between 0 and not set?

Comment (knabben) [reply pkg/kubelet/apis/config/v1beta1/defaults.go:179]: Nice catch for this corner case, moving this to a pointer can solve the problem:

1. If nil (not passed) a default value of 50 is used. - NodeStatusMaxImages:50,
2. If 0 is passed, none images comes from the slice. No .Node.Status.Image
3. If -1 is passed, containerImages list become unlimited.
4. If < -1 is passed, an invalid error is raised.

**Comment by mtaufen** [comment]
/assign @liggitt 
for approval

**Comment by liggitt** [comment]
Are node-published images a thing we intend to continue supporting? I thought experience had shown that complete lists caused performance issues, and incomplete lists were not particularly helpful.

**Comment by liggitt** [comment]
Mechanically, this looks fine. If sig-node and sig-scalability are happy with the node continuing to record images in node status, then this lgtm.

**Comment by wojtek-t** [comment]
> I thought experience had shown that complete lists caused performance issues, and incomplete lists were not particularly helpful.

Incomplete lists sometimes help scheduler to prioritize for image locality. Given how scheduler priorities work (affinity combination of a 10 or so different scores of different priority functions), this is not super predictible anyway, so I've heard users that like this. (and there are not that few users that are not reaching this limit anyway, for which we report all images).

> If sig-node and sig-scalability are happy with the node continuing to record images in node status, then this lgtm.

If we want to stop supporting sending images as part of node status, I'm not opposed, but I would like this to be a formal feature in k8s (which means it will graduate to GA in 1.24 or sth or later).

So in the meantime, I think supporting with this change is what we should do (at least from sig-scalability POV this change seems fine).

@liggitt - WDYT?

**Comment by liggitt** [comment]
if sig-node and sig-scalability are ok with the field, then I don't object

/approve

**Pull request merged**